crates/vm/levm/src/call_frame.rs

```diff
@@ -148,6 +148,11 @@ impl Stack {
             return Err(ExceptionalHalt::StackUnderflow);
         }
 
+        #[expect(unsafe_code, reason = "self.offset always < STACK_LIMIT")]
+        unsafe {
+            assert_unchecked(self.offset < STACK_LIMIT)
+        };
```

Comment:
Let's add a comment explaining that this removes a bound check in the call to `swap`.